spec/controllers/events_controller/index_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::EventsController, type: :controller do
+  describe 'GET #index' do
+    let!(:event_attributes) { %w[id name date description status user_id] }
+    let(:organizer) { create(:user, role: :organizer) }
+    let!(:social_events) { create_list(:event, 3, status: :social, user_id: organizer.id) }
+    let!(:confidential_events) { create_list(:event, 2, status: :confidential, user_id: organizer.id) }
+
+    context 'when user is organizer' do
+      before do
+        @auth_token = organizer.create_new_auth_token
+        request.headers.merge!(@auth_token)
+      end
+
+      it 'returns json with list of all events' do
+        get :index
+        expect(json_response.first.keys).to eq(event_attributes)
+        expect(json_response.count).to eq(social_events.count + confidential_events.count)
+      end
+    end
+
+    context 'when user is attendee' do
+      let(:attendee) { create(:user, role: :attendee) }
+
+      before do
+        @auth_token = attendee.create_new_auth_token
+        request.headers.merge!(@auth_token)
+      end
+
+      it 'returns json with list of social and conforming confidential events' do
+        get :index
+        expect(json_response.first.keys).to eq(event_attributes)
+        expect(json_response.count).to eq(social_events.count)
+      end
+    end
+
+    context 'when unregistered user' do
+      it 'returns json with list of social events' do
+        get :index
+        expect(json_response.first.keys).to eq(event_attributes)
+        expect(json_response.count).to eq(social_events.count)
+        expect(json_response.dig(0, 'status')).to eq('social')
```

Comment:
you can collect event statuses from json response and put 1 expectation that ALL statuses are social